src/live_pixel_validator.mjs

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env node
+import JSON5 from 'json5';
+import { compareVersions, validate as validateVersion } from 'compare-versions';
+
+import { formatAjvErrors } from './error_utils.mjs';
+import { ROOT_PREFIX } from './constants.mjs';
+
+export class LivePixelsValidator {
+    #compiledPixels;
+    #defsVersion;
+    #forceLowerCase;
+
+    undocumentedPixels = new Set();
+    pixelErrors = {};
+
+    constructor(tokenizedPixels, productDef, ignoreParams, paramsValidator) {
```

Comment:
Is there a schema/typedef for `tokenizedPixels` and `productDef`?

Reply:
Same as https://github.com/duckduckgo/pixel-schema/pull/10#discussion_r2002292401

Reply:
`productDef` seems to be quite simple. Would it be possible to just add a quick JSDoc definition of it here?